### PR TITLE
Upgrade apicurio version

### PR DESCRIPTION
**Pull request by carlesarnal**
This PR upgrade the apicurio version fixing this issue https://github.com/quarkusio/quarkus/issues/21534.

### CODE REVIEW

**Comment by carlesarnal** [comment]
cc @cescoffier @Ladicek

**Comment by Ladicek** [comment]
The Apicurio Registry version is defined on a few more places, can you update those too?

**Comment by carlesarnal** [comment]
> The Apicurio Registry version is defined on a few more places, can you update those too?

Done

File: integration-tests/kafka-avro/src/test/java/io/quarkus/it/kafka/KafkaAndSchemaRegistryTestResource.java [hunk 1]
```diff
@@ -29,7 +29,7 @@ public static String getApicurioSchemaRegistryUrl() {
     @Override
     public Map<String, String> start() {
         kafka.start();
-        registry = new GenericContainer<>("apicurio/apicurio-registry-mem:1.2.2.Final")
+        registry = new GenericContainer<>("apicurio/apicurio-registry-mem:2.1.3.Final")
```
Comment (Ladicek) [inline integration-tests/kafka-avro/src/test/java/io/quarkus/it/kafka/KafkaAndSchemaRegistryTestResource.java:32]: Not here, this module intentionally tests with Apicurio Registry 1.x.

Instead, the `kafka-avro-apicurio2` test suite module needs a change in `pom.xml`.

**Review by Ladicek — APPROVED** [review]

**Review by cescoffier — APPROVED** [review]

File: integration-tests/kafka-avro/src/test/java/io/quarkus/it/kafka/KafkaAndSchemaRegistryTestResource.java (hunk 1 above)
Comment (cescoffier) [reply integration-tests/kafka-avro/src/test/java/io/quarkus/it/kafka/KafkaAndSchemaRegistryTestResource.java]: @carlesarnal can you handle Ladislav comment?

**Pull request merged**